### PR TITLE
Fix #186, #184

### LINKED ISSUE #186: Chat should update time passed correctly
image
When a minute passes it shows an hour and it should have been minutes.

### PR DESCRIPTION
**Pull request by luminaxster**

### CODE REVIEW

File: src/chat/chat.js [hunk 1]
```diff
@@ -227,15 +227,23 @@ export class Chat {
...
+    }
+    if(elapsedTimeMinutes >=60  && elapsedTimeMinutes < 120)
     {
       formattedTime = `an hour ago`;
     }
+    else if(elapsedTimeMinutes >=120)
+    {
+      formattedTime = Math.floor(elapsedTimeHours) + ' hours ago';
+    }
     // the same for minutes, hours, days, months, and even years.
```
Comment (luminaxster) [inline src/chat/chat.js]: Let's add: 
if it is less than minute: "just now" (default)
if it is more than 24 hours: "09/27/2016 05:25PM" (the date in this format)

File: src/traceView/expression-data-explorer.js [hunk 2]
```diff
@@ -66,6 +70,25 @@ export class ExpressionDataExplorer{
       $('[data-toggle="tooltip"]').tooltip();
     });
 
+    $editorTooltip.on('shown.bs.popover', function () {
+        //$(".collapsibleListOpen", ".collapsibleListClosed").click(function(){
+          //console.log("#editorTooltipContent.style.width");
+      //});
+
+      let $popoverContentTreeViewContainer = $("#" + $(this).attr("aria-describedby")+".collapsibleListOpen");
```
Comment (luminaxster) [inline src/traceView/expression-data-explorer.js]: @kenmatsuda: could you help me with the action trigger with collapsible list open and closed? Here.

File: src/traceView/expression-data-explorer.js [hunk 3]
```diff
@@ -66,6 +70,25 @@ export class ExpressionDataExplorer{
...
+      $($popoverContentTreeViewContainer).on("click",function(){
+        console.log("#editorTooltipContent.style.width");
```
Comment (luminaxster) [inline src/traceView/expression-data-explorer.js]: You could inspect the context with: console.log(this, arguments);
With these variables, you can get info about the elements involved in the event and find where the width is.

**Comment by kenmatsuda** [comment]
I feel like theres a bit of a miscommunication. The only thing I was having problem with was trying to get collapsible list open and closed to work with event listener which I couldn't do. When i pulled what you did it still didn't work, and now, I'm super confused as to why you got rid of everything.

**Comment by kenmatsuda** [comment]
I mainly want to get back to where we were when me, you and sravya met last monday, thanks for helping all the time!

**Comment by luminaxster** [comment]
Hi Ken,
What I did was already in the branch since Friday, you can look at the
commit history and rebase to that point in time if you overwrite those
changes. Also, use "git pull origin kenbranch" in the command line
interface to get the latest version. You pushed changes but didn't pull
before.

As a common practice, you should let the branch owner merge your changes
into her/his branch after a review. It took me a while to revert the merge
from kenbranch into rc01 and restore your branch.

The issue about controlling the size of the tooltip is still ongoing, I
added the width as a hint to solve the rest of the puzzle. Please pull to
get the update.

Cheers,
~David

**Comment by kenmatsuda** [comment]
Oh okay, sorry about that! So everytime I make a commit thats good I should notify you? also, happy post thanksgiving!

**Comment by kenmatsuda** [comment]
So before I make any changes i should always do git pull origin kenbranch is that correct?

**Comment by luminaxster** [comment]
Hi Ken, thanks for the good wishes, I hope you had a nice weekend.
Yes, the merges are performed by the branch owner. If it is you branch you are updating, you don't have to. 
In this case, you had to pull because I made an push in your branch to illustrate the issue. You can enable Slack desktop notifications to be aware of such updates live.

**Pull request merged**